developing-fcc-software/FccSoftwareGit.md

```diff
@@ -14,81 +14,148 @@ For a general introduction to git, have a look at these tutorials:
+* -d, for packages compiled with debug symbols
 
--   push your local changes to your fork (2), see [below](#contributing-code) how to create a local branch
+#### Branching
 
-    ```
-    git push origin [NAME_OF_LOCAL_BRANCH]
-    ```
+The development should happen in a dedicated branch of your personal repository (2), forked from the official repository (1). To download a local copy of your personal fork, run the following command:
+```bash
+git clone git@github.com:[YOUR_GITHUB_USER]/key4hep_repo.git
+```
 
-### Contributing code
+Move to the new directory,
+```bash
+cd key4hep_repo
+```
 
--   if you are fixing a bug, first create an issue in the github [issue tracker](https://github.com/HEP-FCC/FCCSW/issues)
--   develop your feature in your local copy (3) on a local branch of your choice, to create a branch do:
+Create a new branch for the feature you are developing. This branch is created now on your local copy of the repository. We will have to update the GitHub repository later.
+```bash
+git checkout -b new_feature_branch
+```
 
-    ```
-    git branch -b [NAME_OF_LOCAL_BRANCH]
-    ```
+Now we are ready to start developing the actual changes.
+
+#### Creating changes
+
+Let's replace locally the official repository in the stack by our local copy. To do so, configure (first line), build and install (second line) your local copy of the repository, and then redefine the bash environmental variables to point to your local copy (third line). This is achieved by the following code:
+```bash
+cmake -B build -S . -D CMAKE_INSTALL_PREFIX=install
+cmake --build build -- -j 4  install
+k4_local_repo
+```
+
+#### Create commits and run tests
+
+Modify files as needed for your development. Ideally, each meaninful change should correspond to a commit. Before adding a new commit, it is recommend to run some kind of test (at least that it compiles and run without errors; running ctest would be highly desirable), otherwise there is a risk of leaving the repository in a so-called `broken state`. The following lines show how to mark for commiting, so called `stage`, the files `file1`, `file2` and `README` (first line), and then creating a commit (second line). The commit message should make the change in the files understandable to other developers.
+```bash
+git add file1 README
+git commit -m "new awesome feature"
+```
+Refer to [this tutorial](https://www.atlassian.com/git/tutorials/saving-changes) further details about commiting in Git.
+
+:::{admonition}
+If the feature is completely new in the repository, it is advisable to add a new ctest. Since there is not a standard in Key4hep, it is easier to ask the maintainer of the corresponding repository how to do it. See the following [link](https://github.com/HEP-FCC/FCCSW/blob/master/doc/AddingTestsToFCCSW.md) for further details
+:::
+
+Once all your changes are ready, please run the ctests. To do so, go to the build directory and run `ctest` as follows.
+```bash
+cd build
+ctest
```

Comment:
Maybe mention `ctest -j N` runs N parallel, depending on the repo it can take forever with one job (k4geo...)

Reply:
I think ctest -j N does not honor the dependency among tests, so for a new person running this, can be misleading (there should not be dependencies among tests in first place, but we know how it works...)